.github/workflows/test-build-spack.yml

```diff
@@ -0,0 +1,94 @@
+name: Build ESMF Using Spack
+
+env:
+  esmf_version: 'esmf@develop'
+  nuopc_app_version: 'develop'
+
+on:
+  workflow_dispatch:
+    inputs:
+      esmf_version: 
+        description: 'ESMF version or tag like esmf@develop or esmf@=8.5.0b23'
+        required: false
+        type: string
+        default: 'esmf@develop'
+      nuopc_app_version:
+        description: 'NUOPC Application Prototypes version or tag like develop or v8.5.0b23'
+        required: false
+        type: string
+        default: 'develop'
+  schedule:
+    - cron: '0 6 * * *'
+    - cron: '0 18 * * *'
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+        compiler: [gcc@11.3.0]
+        esmf: ["${{ inputs.esmf_version || env.esmf_version }}+external-parallelio", "${{ inputs.esmf_version || env.esmf_version }}~external-parallelio"]
```

Comment:
I have no idea what the problem is with finding `env`. But it has also surprised me earlier that we have to use `inputs.esmf_version` here, and not `github.event.inputs.esmf_version`. It's like we are already inside the event namespace - and there's no `env` obviously. Maybe try `github.env.esmf_version`?